lib/MySQL_Monitor.cpp

```diff
@@ -5816,5 +5825,1403 @@ bool DNS_Cache::empty() const {
 	return result;
 }
 
+#define NEXT_IMMEDIATE(new_st) do { async_state_machine_=new_st; goto __again; } while (0)
+
+short MySQL_Monitor_State_Data::next_event(MDB_ASYNC_ST new_st, int status) {
+
+	short wait_events = 0;
+
+	if (status & MYSQL_WAIT_READ)
+		wait_events |= POLLIN;
+	if (status & MYSQL_WAIT_WRITE)
+		wait_events |= POLLOUT;
+	if (status & MYSQL_WAIT_EXCEPT)
+		wait_events |= POLLPRI;
+
+#ifdef DEBUG
+	int fd;
+
+	if (wait_events) {
+		fd = mysql_get_socket(mysql);
+	}
+	else {
+		fd = -1;
+	}
+	proxy_debug(PROXY_DEBUG_NET, 8, "fd=%d, wait_events=%d , old_ST=%d, new_ST=%d\n", fd, wait_events, async_state_machine_, new_st);
+#endif /* DEBUG */
+	
+	async_state_machine_ = new_st;
+	return wait_events;
+}
+
+static int
+mysql_status(short event) {
+	int status = 0;
+	if (event & POLLIN) 
+		status |= MYSQL_WAIT_READ;
+	if (event & POLLOUT) 
+		status |= MYSQL_WAIT_WRITE;
+	if (event & POLLPRI) 
+		status |= MYSQL_WAIT_EXCEPT;
+	return status;
+}
+
+class Monitor_Poll {
```

Comment:
Have removed shrinking of array and added max capacity (number of rows).  We need C-array  for poll, though vector can be used here but would be an overkill.
Have used healthcheck_timeout in poll and moved timeout logic from ping_handler/generic_handler to task_handler.

Reply:
Class was used for reusability.
Resizing/reallocation of actual array is not happening just changing len_ variable.

Reply:
Yes, I know the usage that it has, but I don't see the point for reusability. If you mean to reuse previous code, I don't consider that is "reusing" the previous code since it's not a case of refactor + reuse in several places. This class is _all new code_, which takes inspiration in an old implementation, but doesn't quite fit the problem it's solving, so it's modified and it's not like the old implementation. Since a rework is needed anyway (the batching rework), we can revisit this then, but I would try to something simpler like I was mentioning in the previous messages, and later check if this container is justified, instead of trying to make the container fit the problem.